jac-uk/apply
# #788 Enable application form to be configured

## What's included?
In Admin we can now configure which parts of the application are requested at the initial registration stage. 
This PR ensures we only ask the candidate for the chosen 'parts' when they apply.

## Who should test?
✅ Product owner
✅ Developers

## How to test?
In admin configure an exercise to only ask for part of the application data - see jac-uk/admin/pull/1361
For example you may want to only collect 'Personal Details' and 'Additional Information' during initial registration.
Publish that exercise.
Here we are testing that the Apply process is as expected:
 - The vacancy appears in Vacancies list as expected
 - The vacancy details / advert is as expected
 - Candidate can start an application for the vacancy
   - The task list asks for the expected 'parts'
   - Completing a 'part' displays a DONE badge next to it in the task list
   - The 'Review application' button is disabled until all 'parts' are marked as done
   - The 'Review your application' page displays the correct information (and only that which has been asked for)
 - Candidate can submit their application: they receive a success message with an application reference number

## Risk - how likely is this to impact other areas?
🔴 High risk - this includes a lot of changes to shared code


## Additional information
<image src="https://images.zenhubusercontent.com/5e412648a39708a606c962c9/0aad7863-2b0d-41a5-9e51-a1d68e94d8fc" width="50%">
⬇️ 

<image src="https://images.zenhubusercontent.com/5e412648a39708a606c962c9/6e41ad7e-87e4-4b68-8ce1-52a9f563b3f3" width="50%">
⬇️ 

<image src="https://images.zenhubusercontent.com/5e412648a39708a606c962c9/5145d06e-7a96-4e38-933f-eae589977f6c" width="50%">
⬇️ 

<image src="https://images.zenhubusercontent.com/5e412648a39708a606c962c9/b3ea0cc0-f7da-4271-a48f-fb71920c76aa" width="50%">

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
